src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs

```diff
@@ -5576,34 +5576,72 @@ internal static List<CompletionResult> CompleteMember(CompletionContext context,
-                // a MemberExpressionAst '$xml.$xml', whose parent is still a MemberExpressionAst '$xml.$xml.Save'.
-                // But here we DO NOT want to re-assign 'targetExpr' to be '$xml.$xml'. 'targetExpr' in this case
-                // should be '$xml'.
-                if (targetExpr == null)
+                if (lastAst is ErrorExpressionAst)
                 {
-                    var memberExprAst = (MemberExpressionAst)lastAst.Parent;
-                    targetExpr = memberExprAst.Expression;
+                    // Handles scenarios like $PSVersionTable.PSVersi^.Major.
+                    // where the cursor is moved back to a previous member expression while
+                    // there's an incomplete member expression at the end
+                    targetExpr = parentAsMemberExpression;
+                    do
+                    {
+                        if (targetExpr is MemberExpressionAst memberExpression)
+                        {
+                            targetExpr = memberExpression.Expression;
+                        }
+                        else
+                        {
+                            break;
+                        }
+                    } while (targetExpr.Extent.EndOffset >= context.CursorPosition.Offset);
+
+                    if (targetExpr.Parent != parentAsMemberExpression
+                        && targetExpr.Parent is MemberExpressionAst memberAst
+                        && memberAst.Member is StringConstantExpressionAst stringExpression
+                        && stringExpression.Extent.StartOffset <= context.CursorPosition.Offset)
+                    {
+                        memberName = $"{stringExpression.Value}*";
+                    }
+                }
+                else if (targetExpr is null)
+                {
+                    targetExpr = parentAsMemberExpression.Expression;
+                }
+            }
+            else if (lastAst.Parent is BinaryExpressionAst binaryExpression && context.TokenAtCursor.Kind.Equals(TokenKind.Multiply))
+            {
+                if (binaryExpression.Left is MemberExpressionAst memberExpression)
+                {
+                    targetExpr = memberExpression.Expression;
+                    if (memberExpression.Member is StringConstantExpressionAst stringExpression)
+                    {
+                        memberName = $"{stringExpression.Value}*";
+                    }
                 }
             }
-            else if (lastAst.Parent is BinaryExpressionAst && context.TokenAtCursor.Kind.Equals(TokenKind.Multiply))
+            else if (lastAst.Parent is ErrorStatementAst errorStatement)
             {
-                var memberExprAst = ((BinaryExpressionAst)lastAst.Parent).Left as MemberExpressionAst;
-                if (memberExprAst != null)
+                // Handles switches like:
+                // switch ($x)
+                // {
+                //     'value1'.^
```

Comment:
What is `'value1'`? Can we have working example or more clear comment?

Reply:
Value1 is just a random string so we have something to complete members for. The comment is a fully working example, if you paste it into the console and try to complete where the hat is you will get a result.  
PowerShell allows you to create silly switches like this one:

    switch (1..10)
    {
        'four'.Length {"$_ is Four"}
        default {"$_ is not Four"}
    }

Where you use the result from a method/property as the condition. It's very niche, but if the parser supports it then the completion code should as well.

Reply:
> Value1 is just a random string 

If so please replace with `'RandomStringHere'`. In tests too. This helps code reviewers. Thanks!